sdk/python/kubeflow/katib/api/katib_client.py

```diff
@@ -0,0 +1,291 @@
+# Copyright 2019 The Kubeflow Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import multiprocessing
+import time
+
+from kubernetes import client, config
+
+from kubeflow.katib.constants import constants
+from kubeflow.katib.utils import utils
+
+
+class KatibClient(object):
+
+  def __init__(self, config_file=None, context=None, # pylint: disable=too-many-arguments
+               client_configuration=None, persist_config=True):
+    """
+    katibclient constructor
+    :param config_file: kubeconfig file, defaults to ~/.kube/config
+    :param context: kubernetes context
+    :param client_configuration: kubernetes configuration object
+    :param persist_config:
+    """
+    self.in_cluster=None
+    if config_file or not utils.is_running_in_k8s():
+        config.load_kube_config(
+            config_file=config_file,
+            context=context,
+            client_configuration=client_configuration,
+            persist_config=persist_config)
+        self.in_cluster=False
+    else:
+        config.load_incluster_config()
+        self.in_cluster=True
+        
+    self.api_instance = client.CustomObjectsApi()
+
+  def _is_ipython(self):
+    """Returns whether we are running in notebook."""
+    try:
+      import IPython
+      ipy = IPython.get_ipython()
+      if ipy is None:
+        return False
+    except ImportError:
+      return False
+    return True
+  
+  def create_experiment(self, name, namespace=None):
+    """
+    Create the katib experiment
+    :param name: experiment object
```

Comment:
I thick body (or object) is more appropriate than name.

Reply:
changes updated